### PR TITLE
SqlClient minor cleanup to close old issues

### PR DESCRIPTION
**Pull request by Wraith2**
Fixes https://github.com/dotnet/corefx/issues/6134
Fixes https://github.com/dotnet/corefx/issues/5450

Tiny code changes, one to ensure that statistics are handled using try-finally blocks and another to make sure that an AutoResetEvent is cleaned up when the parser instance is disconnected. Auto and manual tests pass. 

/cc the usual suspects @AfsanehR @keeratsingh @saurabh500

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs [hunk 1]
```diff
@@ -975,6 +975,15 @@ internal void Disconnect()
                     _pMarsPhysicalConObj = null;
                 }
             }
+
+            if (_resetConnectionEvent != null)
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: Could be 
```suggestion
            _resetConnectionEvent?.Dispose();
            _resetConnectionEvent = null;
```

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs [hunk 2]
```diff
@@ -569,6 +569,8 @@ override public void Prepare()
 
 
             SqlStatistics statistics = null;
+            try
+            {
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs]: Fix the indentation in the try block and the finally block that has been added.

Comment (Wraith2) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs]: Whoa, i'm sure it wasn't like that when I added it... nevermind, fixed.

**Comment by Wraith2** [comment]
@dotnet-bot test OSX x64 Debug Build
@dotnet-bot test Linux arm64 Release Build

**Comment by AfsanehR** [comment]
@Wraith2  can you also get the latest changes for this pr as well?

**Comment by Wraith2** [comment]
Done, and triggered. I'll work through the other open PR's over the weekend because some will need merge conflict resolution.

**Comment by AfsanehR** [comment]
Merging since the CI failures are not related to the changes made with this PR.

**Pull request merged**